servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,622 @@
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.EventListener;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+import javax.servlet.Filter;
+import javax.servlet.FilterRegistration;
+import javax.servlet.Servlet;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextAttributeEvent;
+import javax.servlet.ServletContextAttributeListener;
+import javax.servlet.ServletContextListener;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRegistration.Dynamic;
+import javax.servlet.ServletRequestAttributeListener;
+import javax.servlet.ServletRequestListener;
+import javax.servlet.SessionCookieConfig;
+import javax.servlet.SessionTrackingMode;
+import javax.servlet.descriptor.JspConfigDescriptor;
+import javax.servlet.http.HttpSessionAttributeListener;
+import javax.servlet.http.HttpSessionIdListener;
+import javax.servlet.http.HttpSessionListener;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.server.servlet.util.MimeMappings;
+import com.linecorp.armeria.server.servlet.util.ServletUtil;
+import com.linecorp.armeria.server.servlet.util.UrlMapper;
+
+import io.netty.handler.codec.http.HttpConstants;
+import io.netty.util.internal.SystemPropertyUtil;
+
+/**
+ * Servlet context (lifetime same as server).
+ */
+public class DefaultServletContext implements ServletContext {
+    private static final Logger logger = LoggerFactory.getLogger(DefaultServletContext.class);
+    /**
+     * Default: 20 minutes.
+     */
+    private int sessionTimeout;
+    private final Map<String,Object> attributeMap = new HashMap<>();
+    private final Map<String,String> initParamMap = new HashMap<>();
+    private final Map<String, ServletRegistration> servletRegistrationMap = new HashMap<>();
+    private final Map<String, ServletFilterRegistration> filterRegistrationMap = new HashMap<>();
+    private final Set<SessionTrackingMode> defaultSessionTrackingModeSet =
+            new HashSet<>(Arrays.asList(SessionTrackingMode.COOKIE,SessionTrackingMode.URL));
+    private final MimeMappings mimeMappings = new MimeMappings();
+    private final ServletEventListenerManager servletEventListenerManager = new ServletEventListenerManager();
+    private final UrlMapper<ServletRegistration> servletUrlMapper = new UrlMapper<>(true);
+    private final UrlMapper<ServletFilterRegistration> filterUrlMapper = new UrlMapper<>(false);
+    private Set<SessionTrackingMode> sessionTrackingModeSet;
+    private String requestCharacterEncoding;
+    private String responseCharacterEncoding;
+    private ClassLoader classLoader = getClass().getClassLoader();
+    private InetSocketAddress serverAddress;
+    private final String contextPath;
+    private final String servletContextName;
```

Comment:
Yes, I know. But I'm not sure if we should support contextPath. If we support context path, all service URLs have to concat context path. Example if context path is (/app), service URL will be (/app/welcome, /app/home) instead of (/wecome, /home) . Should we support context path? How do you think?

Reply:
I think we will support context path later, It is only useful if we deploy multiple apps on one armeria server.

Reply:
If we decide to support contextPath. I will add setContextPath() method to ServletBuilder.

Reply:
> Should we support context path? How do you think?

I don't think so. 😄 Please add setters that you think a user might want to customize.

Reply:
Agree, I will add setters.

Reply:
Also could you add tests for the path we discussed using `ServerExtenstion`?
For example:
When `contextPath = ""` and the servletes are added to the path `"/foo"`, `"/foo/"` and `""`.
When `contextPath = "/foo"` and the servletes are added to the path `"/bar"`, `"/bar/"` and `""`.

Reply:
Agree, I will add tests code.